src/System.Text.RegularExpressions/tests/RegexParserTests.cs

```diff
@@ -0,0 +1,1029 @@
+        [InlineData(@"@""\b\d{2,}\b\D+""", RegexOptions.None, null)]
+        [InlineData(@"@""(00\s){2,4}""", RegexOptions.None, null)]
+        [InlineData(@"@""\b\w*?oo\w*?\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\b\w+?\b""", RegexOptions.None, null)]
+        [InlineData(@"@""^\s*(System.)??Console.Write(Line)??\(??""", RegexOptions.None, null)]
+        [InlineData(@"@""(System.)??""", RegexOptions.None, null)]
+        [InlineData(@"@""\b(\w{3,}?\.){2}?\w{3,}?\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\b[A-Z](\w*?\s*?){1,10}[.!?]""", RegexOptions.None, null)]
+        [InlineData(@"@""b.*([0-9]{4})\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\b.*?([0-9]{4})\b""", RegexOptions.None, null)]
+        [InlineData(@"@""(a?)*""", RegexOptions.None, null)]
+        [InlineData(@"@""(a\1|(?(1)\1)){0,2}""", RegexOptions.None, null)]
+        [InlineData(@"@""(a\1|(?(1)\1)){2}""", RegexOptions.None, null)]
+        [InlineData(@"@""(\w)\1""", RegexOptions.None, null)]
+        [InlineData(@"@""(?<char>\w)\k<char>""", RegexOptions.None, null)]
+        [InlineData(@"@""(?<2>\w)\k<2>""", RegexOptions.None, null)]
+        [InlineData(@"@""(?<1>a)(?<1>\1b)*""", RegexOptions.None, null)]
+        [InlineData(@"@""\b(\p{Lu}{2})(\d{2})?(\p{Lu}{2})\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\bgr[ae]y\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\b((?# case sensitive comparison)D\w+)\s(?ixn)((?#case insensitive comparison)d\w+)\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\{\d+(,-*\d+)*(\:\w{1,4}?)*\}(?x) # Looks for a composite format item.""", RegexOptions.None, null)]
+        // Negative tests
+        [InlineData(@"@""cat([a-\d]*)dog""", RegexOptions.None, RegexParseError.BadClassInCharRange)]
+        [InlineData(@"@""\k<1""", RegexOptions.None, RegexParseError.UnrecognizedEscape)]
+        [InlineData(@"@""\k<""", RegexOptions.None, RegexParseError.UnrecognizedEscape)]
+        [InlineData(@"@""\k""", RegexOptions.None, RegexParseError.MalformedNameRef)]
+        [InlineData(@"@""\1""", RegexOptions.None, RegexParseError.UndefinedBackref)]
+        [InlineData(@"@""(?')""", RegexOptions.None, RegexParseError.InvalidGroupName)]
+        [InlineData(@"@""(?<)""", RegexOptions.None, RegexParseError.InvalidGroupName)]
+        [InlineData(@"@""(?)""", RegexOptions.None, RegexParseError.QuantifyAfterNothing)]
+        [InlineData(@"@""(?>""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?<!""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?<=""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?!""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?=""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?imn )""", RegexOptions.None, RegexParseError.UnrecognizedGro
```

Comment:
what is this?

Reply:
I guess I'm unfamliar with what subtree means in this context.

Reply:
SubTree is something i did in the roslyn tests.  specifically, given any input string we mangle it in many different ways and validate that we dont' crash on any of the mangled names.  Examples of how i mangled:

1.  take the starting string and independently remove each character from it.  i.e.  ```"cat" -> ["at", "ct", "ca"]```.  test all those cases.
2. take the starting string and remove larger and larger prefixes/suffixes.  i.e. ```"cat" -> ["at", "t", "ca", "c"]```.

etc. etc.

basically, a minor form of fuzzing for regexes.  Note: actual fuzzing would be highly appropriate here.  Regex parsing is basically the poster child for that sort of an approach.

Reply:
I agree and believe we should do at least a one-shot fuzzing over our existing engine.

Reply:
Note: if you have an actual fuzzer that .net uses, let me know.  If you create a test suite, i would lke to run it against the roslyn parser.  thanks!